performance-tests/bench_2/worker/main.cpp

```diff
@@ -268,6 +269,79 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[]) {
 
     do_wait(config.start_time, "start");
 
+    Log::log() << "Starting Discovery Check." << std::endl;
+
+    process_start_discovery_time = Builder::get_time();
+
+    if (config.wait_for_discovery) {
+      if (config.wait_for_discovery_seconds > 0) {
+
+        std::chrono::seconds timeoutPeriod(config.wait_for_discovery_seconds);
+        std::chrono::system_clock::time_point timeout_time;
+
+        auto readMap = process.get_reader_map();
+        if (readMap.size() > 0) {
+          typedef std::map<std::string, std::shared_ptr<Builder::DataReader>>::iterator ReadMapIt;
+          std::shared_ptr<Builder::DataReader> dtRdrPtr(nullptr);
+          size_t dr_expected_match_count;
+          size_t dr_match_count;
+
+          for (ReadMapIt it = readMap.begin(); it != readMap.end(); ++it) {
+            dtRdrPtr = it->second;
+            Bench::WorkerDataReaderListener* wdrl = dynamic_cast<Bench::WorkerDataReaderListener*>(dtRdrPtr->get_dds_datareaderlistener().in());
+            std::unique_lock<std::mutex> expected_lock(wdrl->dr_expected_match_cv_mutex);
```

Comment:
If we make the method described above, all this can change to:
```
if (wdrl->wait_for_expected_match(timeout_time)) {
  // Matched Case
} else {
  // Fail Case (This should log some form of the word "Error" so our test scripts can pick up on the issue)
}
```